controllers/bsl.go

```diff
@@ -196,6 +200,33 @@ func (r *DPAReconciler) ReconcileBackupStorageLocations(log logr.Logger) (bool,
 			)
 		}
 	}
+
+	dpaBSLs := velerov1.BackupStorageLocationList{}
+	dpaBSLLabels := map[string]string{
+		"app.kubernetes.io/name":       common.OADPOperatorVelero,
+		"app.kubernetes.io/managed-by": common.OADPOperator,
+		"app.kubernetes.io/component":  "bsl",
+	}
+	err := r.List(r.Context, &dpaBSLs, client.InNamespace(r.NamespacedName.Namespace), client.MatchingLabels(dpaBSLLabels))
+	if err != nil {
+		return false, err
+	}
+	// If current BSLs do not match the spec, delete extra BSLs
+	if len(dpaBSLNames) != len(dpaBSLs.Items) {
+		for _, bsl := range dpaBSLs.Items {
+			if !slices.Contains(dpaBSLNames, bsl.Name) {
+				if err := r.Delete(r.Context, &bsl); err != nil {
+					return false, err
+				}
+				// Record event for BSL deletion
+				r.EventRecorder.Event(&bsl,
+					corev1.EventTypeNormal,
+					"BackupStorageLocationDeleted",
+					fmt.Sprintf("BackupStorageLocation %s in namespace %s was deleted as it was not in spec.", bsl.Name, bsl.Namespace))
```

Comment:
Would be better to be more specific in message? Telling user that the deleted BSL/VSL was created by OADP (not by the user) and telling the object of the spec

A suggestion
> BackupStorageLocation %s **created by OADP** in namespace %s was deleted, because it was not in **DPA** spec.

Reply:
also need to change VSL message